src/snippets/codemirror/snippet_change_spec.ts

```diff
@@ -25,35 +25,48 @@ export class SnippetChangeSpec {
             if (!(text.charAt(i) === "$")) {
                 continue;
             }
-
+    
             let number:number = parseInt(text.charAt(i + 1));
-
+    
             const tabstopStart = i;
             let tabstopEnd = tabstopStart + 2;
             let tabstopReplacement = "";
-
-
+    
+    
             if (isNaN(number)) {
-                // Check for selection tabstops of the form ${0:XXX}
-                if (!(text.charAt(i+1) === "{" && text.charAt(i+3) === ":")) continue;
-
-                number = parseInt(text.charAt(i + 2));
-                if (isNaN(number)) continue;
-
-                // Find the matching }
+                // Check for selection tabstops of the form ${\d+:XXX} where \d+ is some number of
+                // digits and XXX is the replacement string, separated by a colon
+                if (!(text.charAt(i+1) === "{")) continue;
+    
+                // Find the index of the matching closing bracket
                 const closingIndex = findMatchingBracket(text, i+1, "{", "}", false, start + this.insert.length);
-
+                
+                // Create a copy of the entire tabstop string from the document
+                const tabstopString = text.slice(i, closingIndex+1)
```

Comment:
There are a couple missing semicolons here (line 45, as well as lines 51 and 69).